/*
 * This file was automatically generated by EvoSuite
 * Mon May 02 11:46:33 GMT 2022
 */


import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true) 
public class Solution_150_ESTest extends Solution_150_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      Solution_150 solution_150_0 = new Solution_150();
      int[] intArray0 = new int[0];
      // Undeclared exception!
      try { 
        solution_150_0.numberOfLines(intArray0, (String) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("Solution_150", e);
      }
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      Solution_150 solution_150_0 = new Solution_150();
      int[] intArray0 = new int[7];
      intArray0[2] = 100;
      intArray0[6] = 8;
      // Undeclared exception!
      solution_150_0.numberOfLines(intArray0, "cg.`<6");
      try { 
        solution_150_0.numberOfLines(intArray0, "cg.`<6");
        fail("Expecting exception: ArrayIndexOutOfBoundsException");
      
      } catch(ArrayIndexOutOfBoundsException e) {
         //
         // -51
         //
         verifyException("Solution_150", e);
      }
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      Solution_150 solution_150_0 = new Solution_150();
      int[] intArray0 = new int[7];
      // Undeclared exception!
      solution_150_0.numberOfLines(intArray0, "cg.`<6");
      try { 
        solution_150_0.numberOfLines(intArray0, "cg.`<6");
        fail("Expecting exception: ArrayIndexOutOfBoundsException");
      
      } catch(ArrayIndexOutOfBoundsException e) {
         //
         // -51
         //
         verifyException("Solution_150", e);
      }
  }

  @Test(timeout = 4000)
  public void test3()  throws Throwable  {
      Solution_150 solution_150_0 = new Solution_150();
      int[] intArray0 = new int[8];
      int[] intArray1 = solution_150_0.numberOfLines(intArray0, "");
      assertArrayEquals(new int[] {1, 0}, intArray1);
  }
}
